textpipe/doc.py

```diff
@@ -3,38 +3,55 @@
 Clean text, make it readable and obtain metadata from it.
 """
 
-import re
 import functools
+import re
 
-from bs4 import BeautifulSoup
 import cld2
 import spacy
 import textacy
 import textacy.text_utils
+from bs4 import BeautifulSoup
+
+from textpipe.utils import hash_dict
+
 
+class TextpipeMissingModelException(Exception):
+    """Raised when the requested model is missing"""
+    pass
 
-class Doc:
+
+class TextpipeLanguageNotSupportedException(Exception):
+    """Raised when the requested language is not in the list of allowed languages"""
+    pass
+
+
+class Doc:  # pylint: disable=too-many-arguments
     """
     Create a doc instance of text, obtain cleaned, readable text and
     metadata from this doc.
 
     Properties:
     raw: incoming, unedited text
-    clean: string containing the cleaned text
     language: 2-letter code for the language of the text
-    is_detected_language: a boolean indicating if the language was specified
-                            beforehand or detected
     hint_language: language you expect your text to be
+    spacy_nlps: dictionary containing loaded spacy language modules
+    allowed_languages: a tuple with language codes that are allowed in the current pipeline
     """
 
-    def __init__(self, raw, language=None, hint_language='en'):
+    def __init__(self, raw, language=None, hint_language='en', spacy_nlps=None,
+                 allowed_languages=('nl', 'en')):
```

Comment:
What is the rationale behind introducing allowed_languages? SpaCy supports a lot more than only `nl` and `en`, `CLD2` supports 80 languages and `textpipe.doc.Doc` supports all languages in theory.